### PR TITLE
Remove "npm bin" call because it is slow

### PR DESCRIPTION
**Pull request by insidewhy**

### CODE REVIEW

**Review by benjie — CHANGES_REQUESTED** [review]
Excellent idea, well executed. One minor nit: please can you change the /usr/bin fallback to use "npm bin" since it's not always /usr/bin (e.g. /usr/local/bin or an NVM directory or otherwise non-standard path) 🙏

**Comment by insidewhy** [comment]
Good point!

**Comment by insidewhy** [comment]
Pushed over the top of the previous commit now, thanks for your help!

**Comment by benjie** [comment]
Sorry, I should have spotted this earlier (but I was reviewing on my phone...) the functionality of this PR is more limited - previously it would fall back if you didn't have a locally installed version (e.g. if you have a `node_modules` folder but don't have `eslint` installed it'd still fall back to global) whereas now it will fail if you have node_modules but no local eslint even if you have global eslint. I'm thinking you should change it to something more like:

```viml
runtime plugin/neomake-local-eslint.vim
let s:stylelint_path = GetNpmPath('stylelint')
```

I don't imagine that'll be much refactoring? I'm no expert on VimL though...

**Comment by insidewhy** [comment]
@benjie should work now!

File: plugin/neomake-local-eslint.vim [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  let binpath = ''
+  if dir != '/'
+    let binpath = dir . '/node_modules/.bin/' . a:binname
+    if ! filereadable(binpath)
+      let binpath = ''
+    end
+  endif
+
+  if empty(binpath)
+    let binpath = system('echo -n $(npm bin)') . '/' . a:binname
```
Comment (insidewhy) [inline plugin/neomake-local-eslint.vim:19]: `echo -n $(...)` used to avoid newline at end, vim doesn't seem to have a `strip`/`chomp` function.

**Comment by insidewhy** [comment]
Oh it doesn't work, vimscript is so turdy lol.

**Comment by insidewhy** [comment]
Okay now I've tested it properly.

**Review by benjie — APPROVED** [review]

**Comment by benjie** [comment]
LGTM; will look at testing/merging later 👍

**Comment by insidewhy** [comment]
Cool, after that I wanted to add `tslint` (since `stylelint` is also supported and to avoid having to duplicate this `GetNpmBin` across plugins). Would that be cool? I like to use both TypeScript and JavaScript :)

**Comment by insidewhy** [comment]
Alternatively, if you don't want to add `tslint` I could move `GetNpmBin` into a plugin, and then have this plugin depend on that one?

**Comment by benjie** [comment]
I'm totally cool with that; any npm module that's widely used in fact. Can rename it

**Review by benjie — APPROVED** [review]

**Comment by benjie** [comment]
Thanks, I've merged this and then did a little tidying up on master that should make it slightly more efficient.

**Pull request merged**